packages/frontpage/lib/data/db/moderation.ts

```diff
@@ -0,0 +1,23 @@
+import "server-only";
+import { db } from "@/lib/db";
+import * as schema from "@/lib/schema";
+
+export type ModerationEventDTO = {
+  subjectUri: string;
+  subjectDid: string;
+  subjectCollection?: string | null;
+  subjectRkey?: string | null;
+  subjectCid?: string | null;
+  createdBy: string;
+  createdAt: Date;
+  labelsAdded?: string | null;
+  labelsRemoved?: string | null;
+  creatorReportReason?: string | null;
+};
+
+export async function createModerationEvent(
+  moderationEvent: ModerationEventDTO,
+) {
+  await db.insert(schema.ModerationEvent).values(moderationEvent);
```

Comment:
Let's pass each property individually for safety.

Reply:
Typescript allows extra properties on an object in many situations which leads to unsafety. This means that you could pass an object with a property that is unexpected. eg. you could change the schema to include a `foo` column and intentionally not expose it in the `ModerationEventDTO` but because typescript allows additional properties it'll get set in the db anyway.